### ISSUE TITLE
IEX Cloud: Wrong correlation between minute bars and historic prices

### ISSUE THREAD

**Issue by stellarator** [iexg/IEX-API#684]
## Summary

I want to rely on data from 1d chart(s) (my own researches highly depend on 1h stock data, so I have to aggregate 1 min data, but it's OK). But it looks like some divergence (vs EOD) is present(?!)

## Details

This is the very first 'bar' from 1d chart data for `2019-02-12`
```
{
"date":"20190212",
"minute":"09:30",
"label":"09:30 AM",
"high":170.155,
"low":169.9,
"average":170.084,
"volume":6201,
"notional":1054692.925,
"numberOfTrades":49,
"marketHigh":170.2,
"marketLow":169.85,
"marketAverage":170.093,
"marketVolume":761967,
"marketNotional":129604934.4459,
"marketNumberOfTrades":1572,
"open":170.13, // <---
"close":170.08,
"marketOpen":170.13, // <---
"marketClose":170.028,
"changeOverTime":0,
"marketChangeOverTime":0
},
```

And this is aggregate for the mentioned date (2019-02-12) gathered from 1m endpoint (or any other, doesn't matter)
```
{
"date":"2019-02-12",
"open":170.1, // <---
"high":171,
"low":169.7,
"close":170.89,
"volume":22283523,
"uOpen":170.1,
"uHigh":171,
"uLow":169.7,
"uClose":170.89,
"uVolume":22283523,
"change":null,
"changePercent":null,
"label":"Feb 12",
"changeOverTime":0.13926666666666657
}
```

### What actually happened

Prices from the EOD bars correlate perfectly(?) with Yahoo/Google (as an example). But minute bar has different opening price. The questions are: Is it OK? And how should I interpret such cases? And how can I be sure in (data of) other 1d bars?

### Other Information (sample code, links, etc.)

Probably linked to issues #553, #503, and others. But I didn't use v1 endpoints and so could be wrong.

**Comment by joshuablackburn-iex** [comment]
This we need to work with our data partner. The intraday bar data is calculated internally by IEX, and market open and close represent the **official** open or close reported by the listing exchange. The historical close is reported by our partner and adjusted for corporate actions. These two prices can differ based on adjustment, and more commonly, is the result of being the first crossing trade. Since multiple exchanges have open and close cross for a symbol, some vendors (like Yahoo and Google) report that first cross rather than the official open or close. Pro trading terminals will report the official listing exchange open and close, which is the only one that matters for pricing the next day.

We will work with our partner to ensure we get the official open and close, and are investigating ways to deliver both numbers so people can match our data with any other source.

**Comment by stellarator** [comment]
Josh, thank you! I thought about crosses but since my trade account is temporarily on hold I could not verify the idea. Official price (by the listing exchange) is (only) really **important**. Without doubts.

So then the question is transformed into another - could we ask about unadjusted historical (1d and beyond) copy of official OHLC(V)? It would be really great!

**Comment by stellarator** [comment]
@joshuablackburn-iex
I have another question about minute data.

These are two OHLC from [2019-02-21](https://cloud.iexapis.com/beta/stock/aapl/chart/date/20190221?chartByDay=true) and [2019-02-22](https://cloud.iexapis.com/beta/stock/aapl/chart/date/20190222?chartByDay=true) (AAPL):
```
{
  "date":"2019-02-21"
  "open":171.8
  "close":171.06
  "high":172.37
  "low":170.3 // <---
  "volume":17249670
  "uOpen":171.8
  "uClose":171.06
  "uHigh":172.37
  "uLow":170.3 // <---
  "uVolume":17249670
  "change":-0.97
  "changePercent":-0.5639
  "label":"Feb 21, 19"
  "changeOverTime":168.366337
}
```
```
{
  "date":"2019-02-22"
  "open":171.58
  "close":172.97
  "high":173
  "low":171.38 // <---
  "volume":18913154
  "uOpen":171.58
  "uClose":172.97
  "uHigh":173
  "uLow":171.38 // <---
  "uVolume":18913154
  "change":1.91
  "changePercent":1.1166
  "label":"Feb 22, 19"
  "changeOverTime":170.257426
}
```
As usual, I've marked the places that interest me.

This is followed by the data I saved from [1d](https://cloud.iexapis.com/beta/stock/aapl/chart/1d) endpoint (on dates provided).

```
IEX_4D48333344362D52 | 2019-02-22 14:15:00 |   172.51 |   172.55 |  172.48 |   172.509 |      11694 |
IEX_4D48333344362D52 | 2019-02-22 14:16:00 |   172.52 |   172.58 |   164.7 |   172.568 |      36806 |
IEX_4D48333344362D52 | 2019-02-22 14:17:00 |  172.561 |   172.57 |  172.48 |    172.48 |      28915 |
```

```
IEX_4D48333344362D52 | 2019-02-21 15:49:00 |   170.77 |   170.87 |  170.74 |   170.845 |      35289 |
IEX_4D48333344362D52 | 2019-02-21 15:50:00 |   170.85 |   170.89 | 165.536 |    170.78 |      27931 |
IEX_4D48333344362D52 | 2019-02-21 15:51:00 |   170.78 |   170.82 |  170.71 |    170.73 |      42012 |
```
These numbers are **market** OHLCV. Please, pay attention to the middle rows and **Low** column. They contain unusual spikes. And as I understand the subject **High**/**Low** prices are the prices at which transactions were made.

My question is: Is the provided data fully correct (and per day OHLCV by IEX, Google, Yahoo, etc ignores such spikes by some reason(s))? Or something is wrong and what should I/We do to get the correct prices?

The question is not idle, because I do not have time to check the received data on everyday basis (and even more so for each stock). I mean that if I pay for the service (and I started doing it), I would like to be completely confident in it :).

**Comment by evancarter-iex** [comment]
@stellarator We are testing some updates to the High/Low data which should go out in the next couple of days. Can you see if this helps with your situation?

**Comment by stellarator** [comment]
@evancarter-iex (cc: @joshuablackburn-iex )
My situation? Let's be clear. The question is not answered. Is provided 1-min data is almost correct (and as I said - is something wrong with the 1d data and data provided by public sources)? Because - (and it's obvious that) one source is definitely wrong. And if this source is 1-min data - am I right that you know about issue and try to fix it? And if so - I want to be able to re-request correct data for the past periods without unnecessary taxing.

Colleagues, I started to pay for the data. I am a supporter of the theory that it would be better to run the beta version of the API without taking money from clients. So that we can help you find all these errors before you go into commercial use. But, sorry, now it is a contract. As a developer, I fully understand the obstacles that stand in your way. But errors need to be corrected (and there should be a mechanism to help clients to fix errors in their processing copies/caches).

What worries me the most is the statement about the high quality of new data. The question - is the data source really cool and are these just processing errors on your part or ...?

**Comment by stellarator** [comment]
@joshuablackburn-iex / @evancarter-iex 
On the eve of release... Are '1 min' bar data production ready? Or still not? And if so do you have estimates?
After #789 I decided to skip a while and another month later ... Where are we now?
